Kernel_23/doc/Kernel_23/Concepts/FunctionObjectConcepts.h

```diff
@@ -6995,7 +6996,81 @@ class ConstructWeightedCircumcenter_3 {
 
 }; /* end Kernel::ConstructWeightedCircumcenter_3 */
 
+/*!
+  \ingroup PkgKernel23ConceptsFunctionObjects
+  \cgalConcept
+
+  \cgalRefines `AdaptableFunctor` (with one argument)
+
+  \sa `CGAL::Weighted_point_2<Kernel>`
+
+*/
+class ConstructWeightedPoint_2 {
+public:
+
+  /// \name Operations
+  /// A model of this concept must provide:
+  /// @{
+
+  /*!
+    introduces a weighted point with Cartesian coordinates
+    \f$ (0,0)\f$ and weight \f$ 0 \f$.
+  */
+  Kernel::Weighted_point_2 operator()(const CGAL::Origin &CGAL::ORIGIN);
+
+ /*!
+    introduces a weighted point with Cartesian coordinates
+    those of \f$ p \f$ and weight \f$ 0 \f$.
+  */
+  Kernel::Weighted_point_2 operator()(const Kernel::Point_2& p);
+
+  /*!
+     introduces a weighted point with Cartesian coordinates
+     those of \f$ p \f$ and weight \f$ w \f$.
+   */
+   Kernel::Weighted_point_2 operator()(const Kernel::Point_2& p, const Kernel::FT w);
+
+  ///@}
 
+}; /* end Kernel::ConstructWeightedPoint_2 */
+
+/*!
+  \ingroup PkgKernel23ConceptsFunctionObjects
+  \cgalConcept
+
+  \cgalRefines `AdaptableFunctor` (with one argument)
+
+  \sa `CGAL::Weighted_point_3<Kernel>`
+
+*/
+class ConstructWeightedPoint_3 {
+public:
+
+  /// \name Operations
+  /// A model of this concept must provide:
+  /// @{
+
+  /*!
+    introduces a weighted point with Cartesian coordinates
+    \f$ (0,0,0)\f$ and weight \f$ 0 \f$.
+  */
+  Kernel::Weighted_point_3 operator()(const CGAL::Origin &CGAL::ORIGIN);
+
+ /*!
+    introduces a weighted point with Cartesian coordinates
+    those of \f$ p \f$ and weight \f$ 0 \f$.
+  */
+  Kernel::Weighted_point_3 operator()(const Kernel::Point_3& p);
```

Comment:
This branch only finishes the implementation of Weight_point_23. Removing the implicit conversion is WIP in https://github.com/afabri/cgal/tree/Kernel_Weighted_point_without_conversion-GF.